### PR TITLE
fix delete button on users page

### PR DESCRIPTION
**Pull request by iryna-borniak**
dev
## ZeroWaste project

* [Project ticket #654](https://github.com/ita-social-projects/ZeroWaste/issues/654)
* [Project ticket #653](https://github.com/ita-social-projects/ZeroWaste/issues/653)

## Code reviewers

- [ ] @loqimean

### Second Level Review

- [x] @obniavko

## Summary of issue

1. The "Delete" button did not work on the Users page of the admin dashboard.
    **Actual result:** Redirect to the User page after clicking on the "Delete" icon.
    **Expected result:** The corresponding User must be removed.
2. The "Ban" button did not work on the Users page of the admin dashboard.
    **Actual result:** Redirect to the User page without changing "Blocked" status after clicking on the "Ban" icon.
    **Expected result:** A "Blocked" status of the corresponding User must be changed on click. The icon must represent current status.

## Summary of change

1. Fixed the "Delete" button and the "Confirm" message for deleting a User.
2. Added flash notification.
3. Fixed the "Ban" button and the "Confirm" message for blocking/unblocking a User.
4. Implemented toggling for the "Lock" icon.
5. Added corresponding messages for en/uk locales.
6. Set title for Users page.
7. Added tests.
8. Converted account/users/index view from slim into erb format.

Screenshot 2023-12-16 022652
Screenshot 2023-12-16 022713
Screenshot 2023-12-16 021725
Screenshot 2023-12-16 020913

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by obniavko — APPROVED** [review]

File: config/environments/test.rb [hunk 1]
```diff
@@ -34,7 +34,7 @@
   config.cache_store = :null_store
 
   # Raise exceptions instead of rendering exception templates.
-  config.action_dispatch.show_exceptions = false
+  config.action_dispatch.show_exceptions = :none
```
Comment (mdotvasylenchuk) [inline config/environments/test.rb:37]: чому тут none?

File: app/views/account/users/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<div class="container home">
```
Comment (mdotvasylenchuk) [inline app/views/account/users/index.html.erb:1]: додай тут - title

File: app/views/account/users/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+              <td><%= user.email %></td>
+              <td><%= user.first_name %></td>
+              <td><%= user.last_name %></td>
+              <td><%= user.last_sign_in_at %></td>
+              <td><%= link_to icon("fa-solid", "eye"), account_user_path(id: user.id) %></td>
+              <td><%= link_to icon("fa-solid", "edit"), edit_account_user_path(id: user.id) %></td>
+              <td>
+                <%= link_to(icon("fa-solid", user.blocked? ? "lock" : "lock-open"),
+                            account_user_path(id: user, user: { blocked: user.blocked? ? false : true }),
+                            onclick: "alert('Are you sure?')", method: :patch) %>
```
Comment (mdotvasylenchuk) [inline app/views/account/users/index.html.erb:36]: розпиши трохи детальніше повідомлення

File: app/views/account/users/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+              </td>
+              <td>
+                <%= button_to account_user_path(id: user), method: :delete,
+                              data: { turbo_confirm: t(".confirm_delete") } do %>
+                  <i class="fa fa-trash mx-2"></i>
```
Comment (mdotvasylenchuk) [inline app/views/account/users/index.html.erb:41]: тим більше тут юзається і18н а вище ні, так розумію адмінка теж мультимовна, тому виправ кнопку вище

File: app/views/account/users/index.html.erb (hunk 3 above)
Comment (iryna-borniak) [reply app/views/account/users/index.html.erb:36]: Ця логіка повністю перероблена і закривається ПР * [#702](https://github.com/ita-social-projects/ZeroWaste/pull/702/files)

File: app/views/account/users/index.html.erb (hunk 2 above)
Comment (iryna-borniak) [reply app/views/account/users/index.html.erb:1]: В ПР * [#702](https://github.com/ita-social-projects/ZeroWaste/pull/702/files) застосовується caption (стилізований, як заголовок)

File: config/environments/test.rb (hunk 1 above)
Comment (iryna-borniak) [reply config/environments/test.rb:37]: Screenshot 2023-12-24 183950

File: app/views/account/users/index.html.erb (hunk 3 above)
Comment (loqimean) [reply app/views/account/users/index.html.erb:36]: більше того, логіка заблокування в хелпері має бути, а онклік замінити ПОТРІБНО  на турбо конфірм

Comment (loqimean) [reply app/views/account/users/index.html.erb:36]: локалізація теж тре

**Review by loqimean — CHANGES_REQUESTED** [review]

File: app/views/account/users/index.html.erb (hunk 3 above)
Comment (iryna-borniak) [reply app/views/account/users/index.html.erb]: Done

File: app/views/account/users/index.html.erb (hunk 4 above)
Comment (iryna-borniak) [reply app/views/account/users/index.html.erb:43]: Done

**Review by loqimean — CHANGES_REQUESTED** [review]

File: app/helpers/account/users_helper.rb
[the remark this answers is not in the document]
Comment (iryna-borniak) [reply app/helpers/account/users_helper.rb:1]: done

File: app/views/account/users/index.html.erb (hunk 2 above)
Comment (iryna-borniak) [reply app/views/account/users/index.html.erb:1]: done

**Pull request merged**